index.bs

```diff
@@ -696,6 +698,39 @@ authorizing an authenticator with which to complete the operation.
 </div>
 
 
+### Signal platform authenticator capability - isPlatformAuthenticatorReady() method ### {#isPlatformAuthenticatorReady}
+
+<div link-for-hint="WebAuthentication/isPlatformAuthenticatorReady">
+This method is used to signal users whether they can log in to the [RP] with the <a>platform-attached</a> authenticator. 
+The method gives [RP] the ability to detect whether the client is capable of registering credentials with 
+<a>platform authenticators</a> while giving users control over whether they want to share this information. When the 
+method is invoked, the <a>client</a> will search for the existence of an <a>platform authenticator</a> capable of 
+registering credentials. Upon finding one, the <a>client</a> will prompt the user for whether the user is interested
+to register a credential with the authenticator. If so, the platform will redirect the user to a new tab to register 
+the credentials. 
+
+When this method is invoked, the user agent MUST execute the following algorithm: 
+
+1. Let |promise| be a [=a new Promise=] and return the |promise|.
+
+1. Let |isInterested| be a [=boolean=] value. 
+
+1. For all |authenticators| currently available on this platform, if there exists an |authenticator| with 
+    the value of "platform" for its attachment modality, the <a>user agent</a> may take this opportunity to 
+    ask the user whether the user is interested in registering a credential with a <a>platform authenticator</a>.
+    The <a>user agent</a> should show three optional actions to the user: 
```

Comment:
A few comments:
- This seems to go too far in specifying UI, IMO. For example, it implies there is necessarily a dialog box.
- It's not clear to me why this promise always needs to return. What harm is done by just letting it hang out there and never resolving it? You may have to do that anyways in the case there is no authenticator.
- Why are there exactly these three options? I would prefer for this to be in informative text (e.g. a note below the actual normative steps) rather than here. There may be a smaller or larger number of options that's appropriate, depending on the platform.

It seems to me that this would be much simpler if we only said:

1. Let |promise| be a [=a new Promise=]. Return |promise| and execute the following steps [=in parallel=].

1. Enumerate all the [=platform authenticators=] on this platform. If there are no such authenticators, take no further action and terminate this algorithm.

1. If one or more platform authenticators were found, make a determination of the user's willingness to create a credential for this [RP] on such an authenticator. Factors used in this determination may include, for example:
    - Whether the user is running in private or incognito mode
    - Whether the user has configured the browser to never create such credentials
    - Whether the user has on a previous occasion expressed an unwillingness to create a new credential for this [RP], either through configuration or by declining a user interface prompt.
    - The user's explicitly stated intention, as obtained through a browser-specific user interface prompt.

1. If the above procedure determines that the user is not willing to create a new credential on a [=platform authenticator=], take no further action and terminate this algorithm. Otherwise, resolve |promise| with a value of `True`.